robot_smach_states/src/robot_smach_states/startup.py

```diff
@@ -162,7 +165,15 @@ def process_scan(self, laser_upside_down, laser_yaw, door_open, scan_msg):
         if front_index < 2 or front_index > number_beams - 2:
             raise IndexError("Base laser can't see in front of the robot")
 
-        ranges_at_center = scan_msg.ranges[front_index - 2:front_index + 2]  # Get some points around the middle
+        for range_width in range(2, 10):
+            if front_index < range_width or front_index > number_beams - range_width:
+                break
+
+            ranges_at_center = scan_msg.ranges[front_index - range_width:front_index + range_width]  # Get some points around the middle
+            rospy.logdebug(f"{ranges_at_center=}")
+            if not all(map(math.isnan, ranges_at_center)):
+                break
+
```

Comment:
Here we go again. But I'll say fine for now and add it to my list of things to do better at some point.